src/Symfony/Component/Translation/Catalogue/AbstractOperation.php

```diff
@@ -17,38 +17,69 @@
 /**
  * Base catalogues binary operation class.
  *
+ * A catalogue binary operation performs operation on
+ * source (the left argument) and target (the right argument) catalogues and
+ * stores the following results:
+ *
+ * Messages: also called 'all', are valid messages for the given domain after the operation is performed.
+ *
+ * New Messages: also called 'new' (new = all ∖ source = {x: x ∈ all ∧ x ∉ source}).
+ *
+ * Obsolete Messages: also called 'obsolete' (obsolete = source ∖ all = {x: x ∈ source ∧ x ∉ all}).
+ *
+ * Result: also called 'result', is the resulting catalogue for the given domain that holds the same messages as 'all'.
```

Comment:
this is already documented in the interface. No need to duplicate it here (duplication of documentation makes maintenance harder as we have to remember to update it everywhere)